packages/server/graphql/mutations/helpers/notifications/MSTeamsNotifier.ts

```diff
@@ -43,6 +44,45 @@ const notifyMSTeams = async (
 }
 export type MSTeamsNotificationAuth = IntegrationProviderMSTeams & {userId: string}
 
+const createTeamPromptMeetingTitle = (meeting: Meeting) => `*${meeting.name}* is open 💬`
+
+const genericMeetingTitle = `Meeting Started 👋`
+
+const MeetingTypeTitleLookup: Record<MeetingTypeEnum, ((meeting: Meeting) => string) | string> = {
+  action: genericMeetingTitle,
+  poker: genericMeetingTitle,
+  retrospective: genericMeetingTitle,
+  teamPrompt: createTeamPromptMeetingTitle
+}
+
+const createGenericMeetingAction = (meetingUrl: string) => {
+  const joinMeetingAction = new AdaptiveCards.OpenUrlAction()
+  joinMeetingAction.title = 'Join Meeting'
+  joinMeetingAction.url = meetingUrl
+  joinMeetingAction.id = 'joinMeeting'
+
+  return joinMeetingAction
+}
+
+const createTeamPromptMeetingAction = (meetingUrl: string) => {
+  const joinMeetingAction = new AdaptiveCards.OpenUrlAction()
+  joinMeetingAction.title = 'Submit Response'
+  joinMeetingAction.url = meetingUrl
+  joinMeetingAction.id = 'submitResponse'
+
+  return joinMeetingAction
+}
```

Comment:
+1 It seems like a lot of this config is duplicated across Teams, Slack, and Mattermost.  Would it make sense to refactor this into a separate util?  I.e. create an abstraction of "meeting notification" that returns title, URL, button text, etc., and then the individual notifiers could just use that abstracted config instead of doing meeting-specific things for each notification integration.

This would likely reduce shotgun-surgery when creating new meeting types (only change the config, vs changing each notifier) and simplify building new notifier integrations.

Reply:
agreed! will create a refactor task for that.